### DISCUSSION TITLE
K3s setup with signed RSA certs.

### DISCUSSION THREAD

**Discussion by manishm91** [k3s-io/k3s#5239] [Q&A]
Hi,
Trying to set up a K3s server which a client application can connect and do https to.
The client is provided with the k3s.yaml configuration file and it reads up the certs and tokens from this file and tries to connect. However, the client has a limitation in that it is unable to parse EC certs or tokens, and by default the k3s has EC certs. A possible workaround would probably be to generate my own certs, get it signed by some CA and share it with client or install k3s with it, but that would be costly. I was wondering if there is any way that K3S can be set up with signed RSA certs instead of EC certs.

Will be thankful for any insights on how to proceed.

**Comment by brandond** [comment]
No, all of our internal certificate issuance code uses EC keys instead of RSA. What crypto lib are you using that doesn't include EC support?
